site/documentation/content/api/credentials/index.md

```diff
@@ -302,6 +302,39 @@ Example:
+The example above does not contain any of the `not-before`, `not-after` and `enabled` properties. The `not-before`
+and `not-after` properties should be omitted if the validity period is the same as the period indicated by the client
+certificate's corresponding properties. It is still necessary to provide a (empty) JSON object in the *secrets* array,
+though.
 {{% /notice %}}
 
+### Asymmetric Key
+
+A credentials type for storing a public key / certificate as used in the [JSON Web Token based Authentication]({{< relref "concepts/device-identity.md#json-web-token-based-authentication" >}}) to authenticate a device.
+
+Example:
+
+~~~json
+{
+  "device-id": "4711",
+  "type": "asymmetric-key",
+  "auth-id": "sensor1",
+  "secrets": [
+    {
+      "alg": "RSA",
+      "key": "-----BEGIN PUBLIC KEY-----MIIBIj...IDAQAB-----END PUBLIC KEY-----"
+    }
+  ]
+}
+~~~
+
+| Name      | Mandatory | JSON Type | Description                                                                                                                                                                                                                                                                                                |
+|:----------|:---------:|:----------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| *type*    |   *yes*   | *string*  | The credential type name, always `asymmetric-key`.                                                                                                                                                                                                                                                         |
+| *auth-id* |   *yes*   | *string*  | The identity that the device should be authenticated as.                                                                                                                                                                                                                                                   |
+| *alg*     |   *yes*   | *string*  | The signing algorithm of the provided key (supported algorithms are `RSA` and `EC`).                                                                                                                                                                       |
+| *key*     |   *yes*   | *string*  | The Base64 encoded bytes representing the public key / certificate wrapped in a header and footer. Header and footer (either `-----BEGIN PUBLIC KEY-----` and `-----END PUBLIC KEY-----` for public keys or `-----BEGIN CERTIFICATE-----` and `-----END CERTIFICATE-----` for certificates) are mandatory. |
```

Comment:
IMHO we should store the encoded public key itself, i.e. 
> The Base64 encoded binary DER encoding of the public key.

Note that this does not include any `-----BEGIN PUBLIC KEY-----` and `-----END PUBLIC KEY-----` delimiters but just the DER encoded bytes ...

Reply:
updated, but this can actually also be an encoded certificate

Reply:
Well, this property can hold what we define it to hold. FMPOV this is
> The Base64 encoded binary DER encoding of the public key.

and nothing else. It is important to be very specific here, otherwise, clients will have a hard time figuring out what to do with the byte array ...

Reply:
ok makes sense, changed accordingly